Wrappers/Python/ccpi/optimisation/functions/FGP_TV.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+#  CCP in Tomographic Imaging (CCPi) Core Imaging Library (CIL).
+
+#   Copyright 2017-2020 UKRI-STFC
+#   Copyright 2017-2020 University of Manchester
+
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from ccpi.optimisation.functions import Function, MixedL21Norm
+from ccpi.optimisation.operators import Gradient
+import numpy 
+import functools
+
+class FGP_TV(Function):
+    
+    r'''Fast Gradient Projection algorithm for Total Variation(TV) Denoising (ROF problem)
+    
+    .. math::  \min_{x} \alpha TV(x) + \frac{1}{2}||x-b||^{2}_{2}
```

Comment:
Optimisation problem should state that a constraint can be used. Maybe x \in C, where C=reals or non-negative orthant, depending on the parameter "nonnegativity".